react/features/calendar-sync/components/ConferenceNotification.native.js

```diff
@@ -0,0 +1,243 @@
+// @flow
+import React, { Component } from 'react';
+import {
+    Dimensions,
+    Text,
+    TouchableOpacity,
+    View
+} from 'react-native';
+import { connect } from 'react-redux';
+
+import { appNavigate } from '../../app';
+import { getURLWithoutParams } from '../../base/connection';
+import { getLocalizedDateFormatter, translate } from '../../base/i18n';
+import { Icon } from '../../base/font-icons';
+
+import styles from './styles';
+
+const ALERT_MILLISECONDS = 5 * 60 * 1000;
+
+type Props = {
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * The URL of the current conference without params.
+     */
+    _currentConferenceURL: string,
+
+    /**
+     * The calendar event list.
+     */
+    _eventList: Array<Object>,
+
+    /**
+     * The height of the screen. Needed to be in props to trigger a re-render.
+     */
+    _screenHeight: number,
+
+    /**
+     * The translate function.
+     */
+    t: Function
+};
+
+type State = {
+
+    /**
+     * The event object to display the notification for.
+     */
+    event?: Object
+};
+
+/**
+ * Component to display a permanent badge-like notification on the
+ * conference screen when another meeting is about to start.
+ */
+class ConferenceNotification extends Component<Props, State> {
+    updateIntervalId: number;
+
+    /**
+     * Constructor of the ConferenceNotification component.
+     *
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            event: undefined
+        };
+
+        this._getContainerStyle = this._getContainerStyle.bind(this);
+        this._maybeDisplayNotification
+            = this._maybeDisplayNotification.bind(this);
+        this._onGoToNext = this._onGoToNext.bind(this);
+    }
+
+    /**
+     * Implements React Component's componentWillMount.
+     *
+     * @inheritdoc
+     */
+    componentWillMount() {
+        this.updateIntervalId = setInterval(
+            this._maybeDisplayNotification,
+            10 * 1000
+        );
+    }
+
+    /**
+     * Implements React Component's componentWillUnmount.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        if (this.updateIntervalId) {
+            clearTimeout(this.updateIntervalId);
```

Comment:
Fun thing about clearTimeout that I like to leverage: it can take in anything and not error. Up to you if you want to just pass it in updateIntervalId without doing a truthy check.

Reply:
Tricky :) Thanks!